python/ray/dataframe/index_metadata.py

```diff
@@ -272,76 +226,61 @@ def squeeze(self, partition, index_within_partition):
+            coord_df_copy = self._coord_df_cache.copy()
 
+        lengths_copy = self._lengths_cache
+        if not isinstance(self._lengths_cache, ray.local_scheduler.ObjectID):
+            lengths_copy = self._lengths_cache.copy()
 
-class _WrappingIndexMetadata(_IndexMetadata):
-    """IndexMetadata implementation for index across a non-partitioned axis.
-    This implementation assumes the underlying index lies across one partition.
-    """
+        index_copy = self._index_cache
+        if self._index_cache is not None:
+            index_copy = self._index_cache.copy()
 
-    def __init__(self, index):
-        """Inits a IndexMetadata from Pandas Index only.
+        return _IndexMetadata(index=index_copy,
+                              coord_df_oid=coord_df_copy,
+                              lengths_oid=lengths_copy)
 
-        Args:
-            index (pd.Index): Index to wrap.
+    def __getitem__(self, key):
+        return self.coords_of(key)
 
-        Returns:
-            A IndexMetadata backed by the specified pd.Index.
-        """
-        self._coord_df = pd.DataFrame(index=index)
-        # Set _lengths as a dummy variable for future-proof method inheritance
-        self._lengths = [len(index)]
+    def first_valid_index(self):
+        return self._coord_df.first_valid_index()
 
-    def coords_of(self, key):
-        """Returns the coordinates (partition, index_within_partition) of the
-        provided key in the index
+    def last_valid_index(self):
+        return self._coord_df.last_valid_index()
+
+    def drop(self, labels, errors='raise'):
+        """Drop the specified labels from the IndexMetadata
 
         Args:
-            key: item to get coordinates of
+            labels (scalar or list-like):
+                The labels to drop
+            errors ('raise' or 'ignore'):
+                If 'ignore', suppress errors for when labels don't exist
 
         Returns:
-            Pandas object with the keys specified. If key is a single object
-            it will be a pd.Series with items `partition` and
-            `index_within_partition`, and if key is a slice it will be a
-            pd.DataFrame with said items as columns.
+            DataFrame with coordinates of dropped labels
         """
-        locs = self.index.get_loc(key)
-        # locs may be a single int, a slice, or a boolean mask.
-        # Convert here to iterable of integers
-        loc_idxs = pd.RangeIndex(len(self.index))[locs]
-        # TODO: Investigate "modify view/copy" warning
-        ret_obj = self._coord_df.loc[key]
-        ret_obj['partition'] = 0
-        ret_obj['index_within_partition'] = loc_idxs
-        return ret_obj
-
-    def groupby(self, by=None, axis=0, level=None, as_index=True, sort=True,
-                group_keys=True, squeeze=False, **kwargs):
-        raise NotImplementedError()
+        # TODO(patyang): This produces inconsistent indexes.
```

Comment:
Are you planning to add this fix in this PR?

Reply:
This was actually silently fixed by the other changes, setting a new DF will implicitly change the index on the metadata object as well. Comment removed.